visualize-admin/visualization-tool
# feat: Improve data download

Closes #1142.

This PR:
- adds "filtered" or "full" suffix to downloaded file name, based on the download type,
- formats the values properly in XLSX download (note: dates are formatted as strings, and not dates, to make it easier to use them in Excel. This means that e.g. year-resolution values are strings, not numbers),
- sorts the downloaded data `byAuto` from left to right, taking each dimension type into account.

### How to test
1. Go to [this link](https://visualization-tool-git-feat-improve-data-download-f-028d49-ixt1.vercel.app/en/create/new?cube=https://energy.ld.admin.ch/sfoe/bfe_ogd84_einmalverguetung_fuer_photovoltaikanlagen/7).
2. Download chart dataset (XLSX).
3. See that the file name includes "filtered" suffix.
4. Open the file and see that data is sorted by year, ascending.
5. See that columns have proper formats (numbers formatted as numbers).
6. Download full dataset (XLSX).
7. See that the file name includes "full" suffix.
8. See that the data is sorted "from left to right", ascending (first priority – year column, second – Kanton, etc). Kantons have their own sorting order (Zurich, Bern, ...) and this is reflected in the data.
9. Follow the same process e.g. on [PROD environment](https://visualize.admin.ch/en/create/new?cube=https://energy.ld.admin.ch/sfoe/bfe_ogd84_einmalverguetung_fuer_photovoltaikanlagen/7).
10. See that the data is not sorted correctly.
